dts/bindings/disk/zephyr,ram-disk.yaml

```diff
@@ -31,5 +32,16 @@ properties:
     type: phandle
     description: |
       Optional phandle to the memory region to be used as a RAM disk,
-      if not used a local buffer is defined for each disk instance.
-      Use it with caution as it makes memory contents easily accessible.
+      if not used, and 'disk-image' property is also not used, a local buffer is
+      defined for each disk instance. Use it with caution as it makes memory
+      contents easily accessible.
+
+  disk-image:
+    type: string
+    description: |
+      Optional path of a disk image, on the build host system, to be used as
+      backing store of a RAM disk. Make sure that 'sector-size' and 'sector-count'
+      are set to correct values such that 'sector-size' matches the corresponding
+      value used in the existing filesystem (if any) on the disk image and the
+      computed disk size (sector-size x sector-count) matches the size of the disk
+      image in bytes.
```

Comment:
this really has no place in dts, it should be added by cmake functions or similar

Reply:
@nordicjm I don't think there is any alternative available for the multi-instance configuration, where you have multiple RAM disk device nodes and they need to use different disk images (as @decsny also confirms [here](https://github.com/zephyrproject-rtos/zephyr/pull/82603#discussion_r1873838709) or where some RAM disks are specified to use disk images whereas other aren't.

Reply:
You would have the dts node name and you can use that from cmake so it would be something along those lines